x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/utils/MlStringsTests.java

```diff
@@ -46,4 +53,21 @@ public void testHasValidLengthForId() {
         assertThat(MlStrings.hasValidLengthForId(randomAlphaOfLength(64)), is(true));
         assertThat(MlStrings.hasValidLengthForId(randomAlphaOfLength(65)), is(false));
     }
+
+    public void testFindMatching_GivenEmptyItems() {
+        assertThat(MlStrings.findMatching(new String[0], Collections.emptySet()), is(empty()));
+    }
+
+    public void testFindMatching_GivenAllPattern() {
+        assertThat(MlStrings.findMatching(new String[] {"_all"}, new HashSet<>(Arrays.asList("a", "b"))), hasItems("a", "b"));
```

Comment:
IIRC `hasItems` is a partial match (i.e. it asserts that the actual collection has expected items and maybe some more).
I think `org.hamcrest.Matchers.contains` should be used instead in order to achieve full match.

Reply:
:+1: